internal/extractors/aez_reader.go

```diff
@@ -36,7 +36,11 @@ func DecodeAEZ(target chan<- common.DataRecord, source <-chan common.DataRecord)
 		case sourcePacket.TMHeader.IsTCVerification():
 			exportable, err = instrumentVerification(sourcePacket.TMHeader.ServiceSubType, buffer)
 		default:
-			err = errors.New("the TMHeader isn't recognized as either housekeeping, transparent or verification data")
+			err = fmt.Errorf(
+				"the TMHeader isn't recognized as either housekeeping, transparent or verification data (Service Type %v, Service Sub Type %v)",
+				sourcePacket.TMHeader.ServiceType,
+				sourcePacket.TMHeader.ServiceSubType,
+			)
 			exportable = nil
 		}
 		if err != io.EOF {

```

Comment:
You haven't changed this, but if `err == io.EOF` then `sourcePacket.Error` will be `nil` or..?

Reply:
(The reason I looked at this was that you changed something here in a previous commit, which you have since undone. I ask since if `err == nil` this if is `true` and `sourcePacket.Error` is explicitly set to `nil`, whereas if it is `io.EOF` I think it will remain unset, so `nil` by default? If so, then all right, but a tiny bit counter-intuitive.)

Reply:
yes for the SID/RID reading, if the buffer doesn't have enough data this seems to ignore the error (but SID/RID) will be nil, I think. And that ought to cause other errors downstream when determining type of packet.

Reply:
in each step if a package has an error it isn't further processed, so it explicitly has value `nil` in each step of processing if that step is performed. And if it isn't set to anything new it ofc remains nil. Not sure how that is counter intuitive. But in general our EOF handling is a bit off, but rather not deal with that here.

Reply:
skrike that. `io.EOF` is an error that signals graceful end of file from a read, the error we were seeing is `io.ErrUnexpectedEOF`.